### PR TITLE
AC-1082 : "Show Me SparkPost" list for onboarding

### PR DESCRIPTION
**Pull request by shrirupa**
AC-1082 : "Show Me SparkPost" list for onboarding

### What Changed
- Added the GuideListItem.
- List shows up when Show Me SparkPost button is clicked.
- Breadcrumb includes "Show Me SparkPost".

### How To Test
 - Enable "messaging_onboarding" ui option for the account.
 - Navigate to /dashboard.
 - Click on "Start Sending" and then on "Show Me SparkPost".
 - check if list is similar to [mocks](https://sparkpost.invisionapp.com/share/DHULLYL45GB#/screens/390961095)
- adding itemCompleted={true} to one of the <GuideListItem> and check if it is similar to mocks. 
- List persists as the active list through session until switching off of guide set explicitly.

### To Do
- [ ] Address feedback.
- [x]  Write tests

### CODE REVIEW

File: src/pages/dashboard/components/GuideListItem.js [hunk 1]
```diff
@@ -0,0 +1,33 @@
...
+      <div className={styles.CheckBoxContainer}>
+        {itemCompleted ? <CheckCircleOutline size={36} className={styles.CheckCircleIcon}/> : <RadioButtonUnchecked size={36}/>}
+      </div>
+      <div className={styles.ListItemContainer}>
+        {children}
+      </div>
+    </Grid.Column>
+    <Grid.Column md={3} xs={12}>
+      <div className={styles.ListActionContainer}>
+        <Button onClick={onClick} color={(!itemCompleted && 'orange') || null}> {name} </Button>
```
Comment (kchu1337) [inline src/pages/dashboard/components/GuideListItem.js]: UX comment: Is the button disabled? Also the button looks more like the one with an outline prop.

File: src/pages/dashboard/components/GuideListItem.module.scss [hunk 2]
```diff
@@ -0,0 +1,37 @@
...
+
+.ListActionContainer{
+  float: right;
+}
+
+.CheckCircleIcon{
+  color: status-color(success);
+}
+
+.GuideListItem{
```
Comment (kchu1337) [inline src/pages/dashboard/components/GuideListItem.module.scss]: UX comment: Looks like the alignment is different in the mocks. A few pixels down and to the right.

File: src/pages/dashboard/components/tests/GettingStartedGuide.test.js [hunk 3]
```diff
@@ -7,7 +7,8 @@ import { ArrowDownward } from '@sparkpost/matchbox-icons';
 describe('GettingStartedGuide', () => {
   const defaultProps = {
     isGuideAtBottom: false,
-    moveGuideAtBottom: jest.fn()
+    moveGuideAtBottom: jest.fn(),
+    storeStepName: jest.fn()
```
Comment (kchu1337) [inline src/pages/dashboard/components/tests/GettingStartedGuide.test.js]: Could check that this function is called when the button is clicked.

File: src/pages/dashboard/components/tests/GuideListItem.test.js [hunk 4]
```diff
@@ -0,0 +1,27 @@
...
+    children: null,
+    action: {
+      name: 'Button',
+      onClick: jest.fn()
+    }
+  };
+
+  const subject = (props) => (shallow(<GuideListItem {...defaultProps} {...props} />));
+
+  it('should render a orange button only when list item is not complete', () => {
```
Comment (kchu1337) [inline src/pages/dashboard/components/tests/GuideListItem.test.js]: Could split each into 2 separate tests.

File: src/pages/dashboard/components/GettingStartedGuide.js [hunk 5]
```diff
@@ -19,69 +20,109 @@ export const GettingStartedGuide = ({ isGuideAtBottom, moveGuideAtBottom }) => {
...
+  const [stepName, setStepName] = useState(storedStepName || 'Features');
   const guideHeadingRef = useRef(null);
   useEffect(() => {
     if (guideHeadingRef.current) { guideHeadingRef.current.focus(); }
   }, [stepName]);
+  const renderBreadCrumbs = () => (<BreadCrumbs>
+    {breadCrumbsItems[stepName].map((item) => (
+      <BreadCrumbsItem
+        key={item}
+        onClick={() => setStepName(item)}
```
[the remark this answers is not in the document]
Comment (shrirupa) [reply src/pages/dashboard/components/GettingStartedGuide.js]: Good catch! missed this.

File: src/pages/dashboard/components/GuideListItem.js (hunk 1 above)
Comment (shrirupa) [reply src/pages/dashboard/components/GuideListItem.js]: Button is not disabled, user should be able to revisit it. But you are right it looks more like the outline prop one.

File: src/pages/dashboard/components/tests/GuideListItem.test.js (hunk 4 above)
Comment (shrirupa) [reply src/pages/dashboard/components/tests/GuideListItem.test.js]: Since its a if-else like condition, I think we should have it together.

**Pull request merged**